### ISSUE TITLE
[SURVEY] Important Test Cases

### ISSUE THREAD

**Issue by t25kim** [lf-edge/edge-home-orchestration-go#211]
I'd like to know core test cases to add on build workflows like below in order to check the riskiness of PR.
Please leave comments on important test cases, for example `common/sigmgr`.

![image](https://user-images.githubusercontent.com/5847128/103764042-1ab2b780-505e-11eb-8285-dfad0537521a.png)

**Comment by tdrozdovsky** [comment]
This is a good practice! I was on vacation and couldn't answer right away. I think everyone will agree that tests run for every PR.
At the end of 2020, I prepared a workflow to automatically run tests on PR. But I discovered that we have a lot of mistakes. I fixed some inaccuracies in the #212 PR. but there are more serious errors for testing in files:
```
src / controller / discoverymgr / mnedc / client / client_test.go
src / controller / discoverymgr / mnedc / clientcontrol_test.go
src / controller / discoverymgr / mnedc / servercontrol_test.go
src / orchestrationapi / orchestration_test.go
src / orchestrationapi / orchestrationapi_test.go.
```
(there may be problems with the architecture, as function calls are looping)

I think that we will fix them in the near future, and when all tests pass successfully, we will add them to the test workflow. And this will allow us to review only successfully tested code, and also raise the level of the best practices of our project.

Thank @t25kim for this proposal, testing is very important for open source projects.

**Comment by t25kim** [comment]
> This is a good practice! I was on vacation and couldn't answer right away. I think everyone will agree that tests run for every PR.
> At the end of 2020, I prepared a workflow to automatically run tests on PR. But I discovered that we have a lot of mistakes. I fixed some inaccuracies in the #212 PR. but there are more serious errors for testing in files:
> 
> ```
> src / controller / discoverymgr / mnedc / client / client_test.go
> src / controller / discoverymgr / mnedc / clientcontrol_test.go
> src / controller / discoverymgr / mnedc / servercontrol_test.go
> src / orchestrationapi / orchestration_test.go
> src / orchestrationapi / orchestrationapi_test.go.
> ```
> 
> (there may be problems with the architecture, as function calls are looping)
> 
> I think that we will fix them in the near future, and when all tests pass successfully, we will add them to the test workflow. And this will allow us to review only successfully tested code, and also raise the level of the best practices of our project.
> 
> Thank @t25kim for this proposal, testing is very important for open source projects.

@tdrozdovsky Thank you for the comment and PR!
I know there are some issues with mnedc due to the refactoring and hope they will be resolved as soon as possible.
How about running some important TCs through workflows at the moment since some TCs are not working correctly?
We can add whole TCs when all TCs work correctly.
